solr/src/main/java/com/yahoo/ycsb/db/SolrDBClient.java

```diff
@@ -0,0 +1,339 @@
+  }
+
+  /**
+   * Update a record in the database. Any field/value pairs in the specified values HashMap will be
+   * written into the record with the specified record key, overwriting any existing values with the
+   * same field name.
+   *
+   * @param table
+   *          The name of the table
+   * @param key
+   *          The record key of the record to write.
+   * @param values
+   *          A HashMap of field/value pairs to update in the record
+   * @return Zero on success, a non-zero error code on error. See this class's description for a
+   *         discussion of error codes.
+   */
+  @Override
+  public Status update(String table, String key, HashMap<String, ByteIterator> values) {
+    try {
+      SolrQuery query = new SolrQuery();
+      query.setQuery("id:" + key);
+      final QueryResponse readResponse = client.query(table, query);
+      if (readResponse.getStatus() != 0) {
+        return checkStatus(readResponse.getStatus());
+      }
+      if ((readResponse.getResults() != null) && (readResponse.getResults().getNumFound() > 0)) {
+        final SolrDocument updatedDoc = readResponse.getResults().get(0);
+
+        for (Entry<String, String> entry : StringByteIterator.getStringMap(values).entrySet()) {
+          updatedDoc.setField(entry.getKey(), entry.getValue());
+        }
+
+        UpdateResponse writeResponse;
+        if (batchMode) {
+          writeResponse = client.add(table, ClientUtils.toSolrInputDocument(updatedDoc),
+              commitTime);
+        } else {
+          writeResponse = client.add(table, ClientUtils.toSolrInputDocument(updatedDoc));
+          client.commit(table);
+        }
+        return checkStatus(writeResponse.getStatus());
+      }
+    } catch (IOException | SolrServerException e) {
+      e.printStackTrace();
+    }
+    return Status.ERROR;
+  }
+
+  /**
+   * Perform a range scan for a set of records in the database. Each field/value pair from the
+   * result will be stored in a HashMap.
+   *
+   * @param table
+   *          The name of the table
+   * @param startkey
+   *          The record key of the first record to read.
+   * @param recordcount
+   *          The number of records to read
+   * @param fields
+   *          The list of fields to read, or null for all of them
+   * @param result
+   *          A Vector of HashMaps, where each HashMap is a set field/value pairs for one record
+   * @return Zero on success, a non-zero error code on error. See this class's description for a
+   *         discussion of error codes.
+   */
+  @Override
+  public Status scan(String table, String startkey, int recordcount, Set<String> fields,
+      Vector<HashMap<String, ByteIterator>> result) {
+    try {
+      Boolean returnFields = false;
+      String[] fieldList = null;
+      if (fields != null) {
+        returnFields = true;
+        fieldList = fields.toArray(new String[fields.size()]);
+      }
+      SolrQuery query = new SolrQuery();
```

Comment:
Does this query on a scan need to be ordered by the id field? Does it matter if the scan doesn't come back ordered? More a question of convention for YCSB implementation of scans.